common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -394,7 +395,11 @@ private static Collation fetchCollation(int collationId) {
         // instance.
         assert (collationId >= 0 && getDefinitionOrigin(collationId)
           == DefinitionOrigin.PREDEFINED);
-        if (collationId == UTF8_BINARY_COLLATION_ID) {
```

Comment:
can we call these it `IMPLICIT_UTF8_BINARY_COLLATION_ID` and `EXPLICIT_UTF8_BINARY_COLLATION_ID`?

Reply:
I agree, maybe better naming would be `UTF8_BINARY_COLLATION` and `DEFAULT_COLLATION` since we can have `UTF8_BINARY_COLLATION` for both implicit and explicit cases, but this default collation would mean that we have non-collated string?

Reply:
There should be no such thing as non-collated string, each string is collated according to some rules/logic. Naming it like this would imply that utf8 binary is not the default collation when it fact it is. WDYT @srielau

Reply:
I agree, but a `StringType` can have this `collationId` even if the collation is not explicit `UTF8_BINARY`, but also when the collation is implicitly `UTF8_BINARY` or even default `UTF8_BINARY`. Since this is the case, we may want to proceed with different naming? @srielau